src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackCommand.cs

```diff
@@ -108,6 +119,17 @@ public static void Register(CommandLineApplication app, Func<ILogger> getLogger)
                     packArgs.NoDefaultExcludes = noDefaultExcludes.HasValue();
                     packArgs.NoPackageAnalysis = noPackageAnalysis.HasValue();
                     packArgs.OutputDirectory = outputDirectory.Value();
+                    if (properties.HasValue())
+                    {
+                        foreach (var property in properties.Value().Split(';'))
+                        {
+                            int index = property.IndexOf('=');
+                            if (index > 0 && index < property.Length)
```

Comment:
No need to check for (index < property.Length) since it can't possibly be after the end of the string.